internal/logging/middleware.go

```diff
@@ -1,16 +1,51 @@
 package logging
 
 import (
+	"strings"
+	"sync"
 	"time"
 
 	"github.com/gin-gonic/gin"
 	"github.com/rs/zerolog"
 )
 
+type Sampler struct {
+	fn       func() zerolog.Sampler
+	samplers map[string]zerolog.Sampler
```

Comment:
Size of this map _may_ be an issue. Need to keep a close eye on service metrics

Reply:
I see why you're nervous, but I think it will be ok for the time being where we have ~20 endpoints 👍